tests/unit/test_util/test_urlUtils.py

```diff
@@ -0,0 +1,57 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2024 NV Access Limited, Noelia Ruiz Martínez, Leonard de Ruijter
+
+"""Unit tests for the urlUtils submodule."""
+
+import unittest
+from utils.urlUtils import isSamePageUrl
+
+
+class TestIsSamePageUrl(unittest.TestCase):
+	def test_samePage_basic(self):
+		self.assertTrue(isSamePageUrl("http://example.com/page#section", "http://example.com/page"))
+
+	def test_samePage_bothHaveFragments(self):
+		self.assertTrue(isSamePageUrl("http://example.com/page#section", "http://example.com/page#main"))
+
+	def test_differentPage(self):
+		self.assertFalse(isSamePageUrl("http://example.com/otherpage#section", "http://example.com/page"))
+
+	def test_noFragment(self):
+		"""URLs without fragments are not considered the same page."""
+		self.assertFalse(isSamePageUrl("http://example.com/page", "http://example.com/page"))
+
+	def test_differentDomain(self):
+		self.assertFalse(isSamePageUrl("http://other.com/page#section", "http://example.com/page"))
+
+	def test_empty_urlOnPage(self):
+		self.assertFalse(isSamePageUrl("", "http://example.com/page"))
+
+	def test_empty_pageUrl(self):
+		self.assertFalse(isSamePageUrl("http://example.com/page#section", ""))
+
+	def test_differentScheme(self):
+		self.assertFalse(isSamePageUrl("http://example.com/page#section", "https://example.com/page"))
```

Comment:
this should maybe be True? shouldn't these both return to the same page? someone may have opened a website directly to http, but the internal links redirect to https

Reply:
I agree with @seanbudd about this question. Let's ask to @coderabbitai

Reply:
I want to modify this, but just when URLs start with http, not ftp, mailto etc.